### PR TITLE
Add example for router agent

### PR DESCRIPTION
**Pull request by DavdGao**
## Description

Add a demonstration example for router agent.

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: README_ZH.md [hunk 1]
```diff
@@ -153,15 +153,13 @@ AgentScope支持使用以下库快速部署本地模型服务。
...
+  - [自定义工具函数](./examples/conversation_with_customized_services/)
+  - [与SoftWare Engineering智能体对话](./examples/conversation_with_swe-agent/)
   - <img src="https://img.alicdn.com/imgextra/i3/O1CN01SFL0Gu26nrQBFKXFR_!!6000000007707-2-tps-500-500.png" alt="new" width="30" height="30"/>[Mixture of Agents算法](https://github.com/modelscope/agentscope/blob/main/examples/conversation_mixture_of_agents/)
   - <img src="https://img.alicdn.com/imgextra/i3/O1CN01SFL0Gu26nrQBFKXFR_!!6000000007707-2-tps-500-500.png" alt="new" width="30" height="30"/>[流式对话](https://github.com/modelscope/agentscope/blob/main/examples/conversation_in_stream_mode/)
   - <img src="https://img.alicdn.com/imgextra/i3/O1CN01SFL0Gu26nrQBFKXFR_!!6000000007707-2-tps-500-500.png" alt="new" width="30" height="30"/>[与CodeAct智能体对话](https://github.com/modelscope/agentscope/blob/main/examples/conversation_with_codeact_agent/)
-
+  - <img src="https://img.alicdn.com/imgextra/i3/O1CN01SFL0Gu26nrQBFKXFR_!!6000000007707-2-tps-500-500.png" alt="new" width="30" height="30"/>[Router Agent](https://github.com/modelscope/agentscope/blob/main/examples/conversation_with_router_agent/)
```
Comment (garyzhang99) [inline README_ZH.md:162]: [Router Agent]->[与Router Agent对话]？

File: examples/conversation_with_router_agent/router_agent.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        self.sys_prompt = sys_prompt.format_map({"name": self.name})
+
+        self.memory.add(Msg(self.name, self.sys_prompt, "system"))
+
+        self.parser = RegexTaggedContentParser(
+            format_instruction="""Respond with specific tags as outlined below:
+
+- When routing questions to agents:
+<thought>what you thought</thought>
+<agent>the agent name, e.g. math, history</agent>
```
Comment (garyzhang99) [inline examples/conversation_with_router_agent/router_agent.py:38]: 这个地方硬编码和main.py里面的比较耦合了

Comment (DavdGao) [reply examples/conversation_with_router_agent/router_agent.py]: DONE

File: README_ZH.md (hunk 1 above)
Comment (DavdGao) [reply README_ZH.md]: DONE

**Review by garyzhang99 — APPROVED** [review]
LGTM

**Pull request merged**